snuba/datasets/entity_subscriptions/entity_subscription.py

```diff
@@ -65,33 +65,26 @@ def validate_query(self, query: Union[CompositeQuery[Entity], Query]) -> None:
 class SessionsSubscription(EntitySubscriptionValidation, EntitySubscription):
     MAX_ALLOWED_AGGREGATIONS: int = 2
 
-    def __init__(self, data_dict: Mapping[str, Any]) -> None:
-        super().__init__(data_dict)
-        try:
-            self.organization: int = data_dict["organization"]
-        except KeyError:
-            raise InvalidQueryException(
-                "organization param is required for any query over sessions entity"
-            )
-
     def get_entity_subscription_conditions_for_snql(
-        self, offset: Optional[int] = None
+        self, offset: Optional[int] = None, organization: Optional[int] = None
     ) -> Sequence[Expression]:
+        if not organization:
+            raise InvalidQueryException("organization param is required")
         return [
             binary_condition(
                 ConditionFunctions.EQ,
                 Column(None, None, "org_id"),
-                Literal(None, self.organization),
+                Literal(None, organization),
             ),
         ]
 
     def to_dict(self) -> Mapping[str, Any]:
-        return {"organization": self.organization}
+        return self.__dict__
```

Comment:
I don't think this is right. This will start returning a bunch of values that weren't being returned before. Really this should return at more an empty dictionary, if organization is being encoded elsewhere.

That being said, it doesn't make a lot of sense to have a function that just returns an empty dictionary. If it doesn't overcomplicate the change, I would remove `to_dict` from the entity subscription completely.

Reply:
Right, organization is already being encoded on the `SubscriptionData`. Was thinking since `entity_subscription` is an attribute of the subscription data, it might make sense to encode that data too. But since this changes the way we do thinks right now, I will remove.